### PR TITLE
Add EDA deployment on OCP

### PR DESCRIPTION
**Pull request by derekwaters**
# What does this PR do?

Add EDA component deployment to the aap_install_ocp role to allow automated deployment of EDA alongside AAP and Hub components.

# How should this be tested?

Adding the aap_ocp_install_eda var definition to an existing use of aap_install_ocp will also deploy the EDA component. An example playbook which installs AAP controller and EDA onto an existing OCP environment is included below:

```
---
- name: Install AAP and EDA on an existing OCP Cluster
  vars:
    aap_ocp_install_connection:
      host: "{{ ocp_host }}"
      username: "{{ ocp_username }}"
      password: "{{ ocp_password }}"
      verify_ssl: false
      validate_certs: false
    aap_ocp_install_namespace: aap
    aap_ocp_install_operator:
      channel: "stable-2.4"
    aap_ocp_install_controller:
      instance_name: demo-aap
      replicas: 1
      link_text: "Ansible Automation Platform"
    aap_ocp_install_eda:
      instance_name: demo-eda
      replicas: 1
      link_text: "Ansible Automation Platform EDA"

  block:
    - name: Install AAP and EDA
      ansible.builtin.include_role:
        name: "infra.aap_utilities.aap_ocp_install"
```

# Is there a relevant Issue open for this?

No issue open for this

# Other Relevant info, PRs, etc

No related info or PRs

### CODE REVIEW

**Comment by branic** [comment]
Hi @derekwaters Thank you for your contribution. I'm starting to look over this PR and test it.

While I in the process of reviewing can you please add a changelog fragment and update the role's readme.

File: roles/aap_ocp_install/tasks/install-eda.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+- name: Create EDA instance
```
Comment (branic) [inline roles/aap_ocp_install/tasks/install-eda.yml:16]: There should be a task to create a namespace if the defined namespace doesn't exist. See the first task in the `install-controller.yml` file for an example. You should be able to copy that task and change the variable names. (I realized in reviewing this PR that the hub install tasks don't do this either so I'll be fixing that with an upcoming PR).

File: roles/aap_ocp_install/tasks/pre-validate-eda.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (branic) [inline roles/aap_ocp_install/tasks/pre-validate-eda.yml:1]: There should be a few more pre-validation tasks such as checking the namespace name and link text if they are defined. Have a look at the checks in the `pre-validate-controller.yml` and use them as a reference to implement them for EDA (not all of them will be applicable to EDA).

**Review by branic — CHANGES_REQUESTED** [review]
Hi @derekwaters I've had some time to do some testing and found a couple of things in addition to the documentation updates that I'd like to see addressed.

Again, thank you for your work on this PR.

File: roles/aap_ocp_install/tasks/install-eda.yml (hunk 1 above)
Comment (derekwaters) [reply roles/aap_ocp_install/tasks/install-eda.yml:16]: Done!

File: roles/aap_ocp_install/tasks/pre-validate-eda.yml (hunk 2 above)
Comment (derekwaters) [reply roles/aap_ocp_install/tasks/pre-validate-eda.yml:1]: Added pre-validation checks for namespace, link text and replicas values.

File: roles/aap_ocp_install/defaults/main.yml [hunk 3]
```diff
@@ -24,4 +24,11 @@
 # aap_ocp_install_hub:
 #   instance_name:
 #   link_text:
+
+# Variable applicable to the EDA installation
+# aap_ocp_install_eda:
+#   namespace:
+#   instance_name:
+#   replicas:
```
Comment (branic) [inline roles/aap_ocp_install/defaults/main.yml]: There is a tailing space that needs to be removed to pass the linting.
```suggestion
#   replicas: 
```
```suggestion
#   replicas:
```

File: roles/aap_ocp_install/tasks/install-eda.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    validate_certs: "{{ aap_ocp_install_connection['validate_certs'] | default(omit) }}"
+    state: present
+    resource_definition: "{{ lookup('ansible.builtin.template', 'namespace.yaml.j2', template_vars=ns_vars) | from_yaml }}"
+    apply: true
+  vars:
+    ns_vars:
+      ns_name: "{{ aap_ocp_install_eda['namespace'] }}"
+  when:
+    - aap_ocp_install_eda['namespace'] is defined
+
```
Comment (branic) [inline roles/aap_ocp_install/tasks/install-eda.yml]: Spaces here also need to be removed here to pass linting.

**Review by branic — CHANGES_REQUESTED** [review]
@derekwaters Thanks for making the requested changes. This looks really good and I haven't hit any issues in my testing. I did notice a couple of linting errors that need to be addressed, so if you wouldn't mind making the requested changes to fix the linting then I'll get this PR merged.

File: roles/aap_ocp_install/defaults/main.yml (hunk 3 above)
Comment (derekwaters) [reply roles/aap_ocp_install/defaults/main.yml]: Corrected

**Comment by derekwaters** [comment]
Apologies @branic I wasn't doing those changes on my normal dev machine and missed my ansible-lint check. Should all be good now. Thanks!

**Review by branic — APPROVED** [review]

**Pull request merged**